### DISCUSSION TITLE
Plugin for `Rotation enabled Text`

### DISCUSSION THREAD

**Discussion by MiltonBalaOfficial** [xournalpp/xournalpp#6152] [Show and tell] [answered]
Sometimes, rotating text is essential, especially when working with graphs such as those in #4614, #4899, and #1566. Although rotated text can be inserted using LaTeX, it isn't the most user-friendly option for general users. I have an idea for a plugin that would convert normal text into "rotation-enabled text." While this wouldn't keep the text as editable text, it would treat the text like strokes, giving the appearance of rotated text. I believe this plugin could be very useful, even though it may not fully replace native support for text rotation.

At this point, it’s important to know whether the rotation property for actual text will be implemented soon. If it is not, I plan to invest some effort into developing the plugin to at least provide a similar experience of rotation-enabled text.

**Comment by MiltonBalaOfficial** [comment]
> it’s important to know whether the rotation property for actual text will be implemented soon.

I was expecting a reply but without waiting much I have developed a plugin to have a output like this:
![image](https://github.com/user-attachments/assets/e0f847f3-fbc3-4cbe-a1c2-b6acc2af03e5)

**Reply by rolandlo** [reply, marked as answer]
That looks nice. How did you generate the rotated text? Did you just create strokes for all letters manually?
As for how long it will take for the rotation property of text getting implement. That's kind of hard to predict. There is PR #5596, but it's not finished and currently not actively developed. It will certainly take more than a couple of days.

**Reply by MiltonBalaOfficial** [reply]
Waiting eagerly for the rotation property of the `Text`, but right now what I can afford is this plugin:  [ConvertTextToRotationEnabledTextStroke](https://github.com/MiltonBalaOfficial/ConvertTextToRotationEnabledTextStroke). Right now it is able to fulfil my needs and may also serve others. Any suggestion is welcome.
